core/include/gnuradio-4.0/BlockTraits.hpp

```diff
@@ -281,20 +281,53 @@ concept can_processMessagesForPortConsumableSpan = requires(TBlock &block, TPort
 template<typename TBlock, typename TPort>
 concept can_processMessagesForPortStdSpan = requires(TBlock &block, TPort &inPort, std::span<const Message> msgSpan) { block.processMessages(inPort, msgSpan); };
 
+// clang-format off
 namespace detail {
+
 template<typename T>
-struct dummy_input_span : std::span<const T> {           // NOSONAR
-    dummy_input_span(const dummy_input_span &) = delete; // NOSONAR
-    dummy_input_span(dummy_input_span &&) noexcept;      // NOSONAR
-    constexpr void consume(std::size_t) noexcept;
+struct dummy_input_span {
+    using value_type = typename std::remove_cv_t<T>;
+    using iterator = typename std::span<const T>::iterator;
+
+    dummy_input_span() = default;
+    dummy_input_span(const dummy_input_span& other) = default;
+    dummy_input_span& operator=(const dummy_input_span& other) = default;
+    dummy_input_span(dummy_input_span&& other) noexcept = default;
+    dummy_input_span& operator=(dummy_input_span&& other) noexcept = default;
+    ~dummy_input_span() = default;
+
+    [[nodiscard]] constexpr iterator begin() const noexcept;
+    [[nodiscard]] constexpr iterator end() const noexcept;
+    operator const std::span<const T>&() const noexcept;
+    operator std::span<const T>&() noexcept;
+    operator std::span<const T>&&() = delete;
+
+    [[nodiscard]] bool consume(std::size_t nSamples) const noexcept;
 };
+static_assert(ConsumableSpan<dummy_input_span<int>>);
 
 template<typename T>
-struct dummy_output_span : std::span<T> {                  // NOSONAR
-    dummy_output_span(const dummy_output_span &) = delete; // NOSONAR
-    dummy_output_span(dummy_output_span &&) noexcept;      // NOSONAR
+struct dummy_output_span {
```

Comment:
Naming: `DummyPublishableSpan`? The current naming makes it hard to understand the templates since it is not clear whihc concept is mocked here.

Reply:
Done.